ezyrb/ann.py

```diff
@@ -25,9 +25,47 @@ class ANN(Approximation):
     >>> print(y_pred)
     """
         
-    def __init__(self):
-        self.trained_epoch = 0   
-        self.criterion = torch.nn.MSELoss()
+    def __init__(self, layers, function, stop_training, loss=None):
+
+        if loss is None: loss = torch.nn.MSELoss
+        if optimizer is None: optimizer = torch.optim.Adam
+        if not isinstance(function, list): # Single activation function passed
+            function = [function] * (len(layers)-1)
+
+
+        self.layers = layers
+        self.function = function
+        self.loss = loss
+        self.stop_training = stop_training
+
+        self.loss_trend = []   
+        self.model = None
+
+    def _convert_numpy_to_torch(self, array):
+        """
+        Converting data type.
+        TODO: type should be not forced to `float`
+        """
+        return torch.from_numpy(array).float()
+
+    def _convert_torch_to_numpy(self, tensor):
+        """
+        Converting data type.
+        """
+        return tensor.detach().numpy()
+
+    def _build_model(self):
```

Comment:
"points" and "values" should to be considered as input parameters of the method.